server/src/controllers/users.controller.ts

```diff
@@ -37,6 +48,14 @@ export class UsersController {
     };
   }
 
+  @Post('avatar')
+  @UseGuards(JwtAuthGuard)
+  @UseInterceptors(FileInterceptor('file'))
+  async addAvatar(@Req() req, @UploadedFile() file: Express.Multer.File) {
+    if (!file || !file?.buffer) throw new BadRequestException('File cannot be empty');
```

Comment:
could use dto directly for this to validate here and then pass it onto the service